[chenle228] iP

chenle228 opened the pull request:
# Lolo
>  “The secret of getting ahead is getting started.” – [Mark Twain](https://www.goodreads.com/quotes/219455-the-secret-of-getting-ahead-is-getting-started-the-secret)

Lolo is a Java-based chatbot that helps you manage your tasks with ease. It's:

- Text-based for simplicity
- Easy to learn with intuitive commands
- ~~FAST~~ _SUPER FAST_ and responsive ⚡

All you need to do is:
1.  Download it from [here](https://github.com/chenle228/ip/releases/tag/v0.1).
2.  Double-click to launch.
3.  Add your tasks.
4.  Let Lolo handle the rest for you! 😀

And yes, it's **FREE**! 🎉

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [x] Searching tasks by keywords
- [ ] Reminders (coming soon)
If you're a Java developer, you can also use Lolo to brush up on your skills. Here's an example of the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

With Lolo, you can add various types of tasks like ToDo, Deadline, and Event. You can also search for tasks containing specific keywords using the find command.

What are you waiting for? Get started now!🙂

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

chenle228 committed:
Implement an initial skeletal version of the bot

chenle228 committed:
Improve the skeletal version with Echo

chenle228 committed:
Implement add and list tasks feature

chenle228 committed:
Implement mark and unmark features

chenle228 committed:
Implement Tracking for ToDos, Events, Deadlines

chenle228 committed:
Implement Automated Text UI Testing

chenle228 committed:
Handle incorrect user inputs

chenle228 committed:
Implement delete task feature and update test

chenle228 committed:
Save the tasks in the hard disk

chenle228 committed:
Merge branch 'branch-Level-7'

chenle228 committed:
Parse dates and times

chenle228 committed:
Merge branch 'branch-Level-8'

chenle228 committed:
Refactor the code to make it more OOP

chenle228 committed:
Merge branch 'branch-A-MoreOOP'

chenle228 committed:
Organise code into packages

chenle228 committed:
Merge branch 'branch-A-Packages'

chenle228 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

chenle228 committed:
Set up gradle for the project

chenle228 committed:
Add checkstyle files, run, test, and clean project

chenle228 committed:
Merge branch 'branch-A-Gradle'

chenle228 committed:
Add JUnit Tests for Storage and TaskList classes

chenle228 committed:
Merge branch 'branch-A-JUnit'

chenle228 committed:
Update build.gradle with shadow plugin config

chenle228 committed:
Merge branch 'branch-A-Jar'

chenle228 committed:
Add JavaDoc comments to the code

chenle228 committed:
Merge branch 'branch-A-JavaDoc'

chenle228 committed:
Remove unused import statements to follow standard

chenle228 committed:
Merge branch 'branch-A-CodingStandard'

chenle228 committed:
Implement search-by-keyword feature in task list

chenle228 committed:
Merge branch 'branch-Level-9'

jolwnn commented on src/main/java/lolo/storage/Storage.java:
```diff
@@ -0,0 +1,120 @@
+                            break;
+                        default:
+                            throw new LoloException("Unknown task type in file.");
+                    }
```
Should this indentation be removed? According to Java coding standard, there should be no indentation for case clauses.

yikyak02 replied:
i agree, perhaps editing it would help better comply with java coding standards

jolwnn commented on src/main/java/lolo/command/FindCommand.java:
```diff
@@ -0,0 +1,23 @@
+class FindCommand extends Command {
+    private String keyword;
+
+    public FindCommand(String keyword) {
```
Maybe add a method header here? You added method headers for all the other methods though, I think, so LGTM!

jolwnn commented on src/main/java/lolo/task/TaskList.java:
```diff
@@ -0,0 +1,145 @@
+     * @param index The index of the task to be retrieved.
+     * @return The task at the specified index.
+     */
+    public Task get(int index) {
```
Perhaps you could use a more specific method name?

jolwnn commented on src/main/java/lolo/task/Task.java:
```diff
@@ -0,0 +1,64 @@
+ */
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
```
Its great to see that you followed the boolean naming convention

jolwnn commented on src/main/java/lolo/Ui.java:
I like how clean this bit of code is 👍 The method names are all descriptive and follow the coding standard!

jolwnn commented on src/main/java/lolo/task/TaskList.java:
```diff
@@ -0,0 +1,145 @@
+ * and retrieving tasks based on specific criteria.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
```
I think its great that you used plural for tasks!

jolwnn reviewed (commented):
Overall, the code LGTM! :))

yikyak02 commented on src/main/java/lolo/command/MarkCommand.java:
```diff
@@ -0,0 +1,42 @@
+     *
+     * @param taskNumber The index of the task to be marked as done.
+     */
+    public MarkCommand(int taskNumber) {
```
this method name could be rewritten in Camelcase :)

yikyak02 commented on src/main/java/lolo/command/UnmarkCommand.java:
```diff
@@ -0,0 +1,41 @@
+ * This command updates the status of a task and interacts with
+ * the user interface and storage to reflect the change.
+ */
+class UnmarkCommand extends Command {
```
this method name could be rewritten in Camelcase :)

yikyak02 commented on src/main/java/lolo/task/ToDo.java:
```diff
@@ -0,0 +1,40 @@
+package lolo.task;
```
good job on the use of egyptian style indentations!

yikyak02 commented on src/main/java/lolo/task/Task.java:
```diff
@@ -0,0 +1,64 @@
+     *
+     * @return The status icon of the task.
+     */
+    public String getStatusIcon() {
```
good and descriptive method name!

yikyak02 reviewed (commented):
LGTM, keep up the work!

chenle228 committed:
Implement the GUI with with JavaFX

chenle228 committed:
Merge branch 'branch-Level-10'

chenle228 committed:
Update README.md

chenle228 committed:
Update README.md

chenle228 committed:
Lolo class: Add assertions to document assumptions

The Lolo class currently lacks explicit checks for key assumptions in the code, such as ensuring non-null values for critical variables like filePath, tasks, and input.

This can lead to unexpected errors during execution if these assumptions are violated, making it harder to trace bugs and understand the root cause.

Let's:

Add assertions to validate the filePath in the constructor to ensure it's not null.
Add assertions in getResponse to check that the input string, parsed command, and response are all non-null.
Add an assertion in main to ensure that a file path argument is provided during initialization.
Using assertions helps make assumptions clear and catches potential bugs early, without impacting performance in production since assertions can be disabled.

The chosen assertions make sure the core logic operates on valid data and prevent potential null pointer exceptions or incorrect behavior. This approach is preferable to adding defensive checks because these are internal assumptions rather than user-facing error handling.

chenle228 committed:
Remove unused methods to improve code quality

chenle228 committed:
Merge pull request #1 from chenle228/branch-A-Assertions

Lolo class: Add assertions to document assumptions

chenle228 committed:
Merge pull request #2 from chenle228/branch-A-CodeQuality

Remove unused methods to improve code quality

chenle228 committed:
Implement tagging feature

chenle228 committed:
Merge branch 'branch-C-Tagging'

chenle228 committed:
Update README.md

chenle228 committed:
Refactor code and shorten methods in parser class

chenle228 committed:
Merge branch 'master' of https://github.com/chenle228/ip

chenle228 committed:
Update javadoc comments in parser class

chenle228 committed:
Refactor code to shorten methods in storage class

chenle228 committed:
Update javadoc comments in lolo class

chenle228 committed:
Delete old binary file of duke

chenle228 committed:
Add a representative screenshot of the product

chenle228 committed:
Add a User Guide

chenle228 committed:
Merge pull request #4 from chenle228/branch-A-UserGuide

Add a user guide for the project

chenle228 committed:
Add javadoc comments for handling todo command

chenle228 committed:
Add javadoc comments for handling deadline command

chenle228 committed:
Add javadoc comments for handling event command

chenle228 committed:
Add javadoc comments for handling mark command

chenle228 committed:
Add javadoc comments for handling unmark command

chenle228 committed:
Add javadoc comments for handling delete command

chenle228 committed:
Add javadoc comments for list on date command

chenle228 committed:
Add javadoc comments for handling find command

chenle228 committed:
Add javadoc comments for handling tag command

chenle228 committed:
Update user guide.md

chenle228 committed:
Update user guide.md

chenle228 committed:
Update user guide.md

chenle228 committed:
Update README.md

chenle228 committed:
Update README.md

chenle228 committed:
Remove fixed file path from main class

chenle228 committed:
Merge branch 'master' of https://github.com/chenle228/ip

chenle228 committed:
Update build.gradle file

chenle228 committed:
Delete docs/Ui.png

chenle228 committed:
Add a representative screenshot of the product

chenle228 committed:
Merge pull request #5 from chenle228/branch-A-Release

Add a representative screenshot of the product

chenle228 committed:
Delete docs/Ui.png

chenle228 committed:
Add a representative screenshot of the product

chenle228 committed:
Update input.txt

chenle228 committed:
Merge branch 'master' of https://github.com/chenle228/ip

chenle228 committed:
Update README.md

Added expected output for every command
